### PR TITLE
Enable dependabot updates for GHA

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Adds version updates for GHA via dependabot.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Review by marcoesters — COMMENTED** [review]
I suggest adding the exact version number as a comment behind the hashes so that it's easy to see which version is used.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -33,25 +33,25 @@ jobs:
...
-        uses: actions/checkout@v3
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
 
       - name: Timestamp
         run: echo "TIMESTAMP=$(date -u "+%Y%m")" >> $GITHUB_ENV
         shell: bash
 
       - name: Cache conda
-        uses: actions/cache@v3
+        uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c
```
Comment (kenodegard) [inline .github/workflows/tests.yml:48]:
```suggestion
        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -33,25 +33,25 @@ jobs:
...
         with:
           path: ~/conda_pkgs_dir
           key: ${{ runner.os }}-conda-content-trust-${{ env.TIMESTAMP }}
 
       - name: Setup miniconda
-        uses: conda-incubator/setup-miniconda@v2
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca
```
Comment (kenodegard) [inline .github/workflows/tests.yml:54]:
```suggestion
        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -70,7 +70,7 @@ jobs:
         run: pytest --cov=conda_content_trust
 
       - name: Upload coverage
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@ab904c41d6ece82784817410c45d8b8c02684457
```
Comment (kenodegard) [inline .github/workflows/tests.yml:73]:
```suggestion
        uses: codecov/codecov-action@5ecb98a3c6b747ed38dc09f787459979aebb39be # v4.3.1
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -99,14 +99,14 @@ jobs:
...
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
         with:
           ref: ${{ github.ref }}
           clean: true
           fetch-depth: 0
 
       # Explicitly use Python 3.11 since each of the OSes has a different default Python
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236
```
Comment (kenodegard) [inline .github/workflows/tests.yml:109]:
```suggestion
      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
```

**Review by marcoesters — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**